caikit_nlp/modules/text_classification/sequence_classification.py

```diff
@@ -0,0 +1,235 @@
+            ClassificationResult
+        """
+        scores_dict = self._get_scores(text)
+        # Re-organize scores_dict - for one text, this is just the first score
+        return SequenceClassification._process_predictions(scores_dict, text_idx=0)
+
+    def run_batch(self, texts: List[str]) -> List[ClassificationResult]:
+        """Run the sequence classification on batch, truncates sequences too long for model
+
+        Args:
+            text: List[str]
+                Input strings to be classified
+
+        Returns:
+            List[ClassificationResult]
+        """
+        scores_dict = self._get_scores(texts)
+        num_texts = len(texts)
+
+        # Re-organize scores_dict for each example
+        # We could eventually consider whether or not to sort classifications by scores
+        # but avoiding this prescription here for now
+        classification_predictions = []
+        for text_idx in range(num_texts):
+            classification_prediction = SequenceClassification._process_predictions(
+                scores_dict, text_idx
+            )
+            classification_predictions.append(classification_prediction)
+        return classification_predictions
+
+    def save(self, model_path: str):
+        """Save model in target path
+
+        Args:
+            model_path: str
+                Path to store model artifact(s)
+        """
+        self.tokenizer.save_pretrained(model_path)
+        self.model.save_pretrained(model_path)
+
+    @classmethod
+    def load(cls, model_path: str) -> "SequenceClassification":
+        """Load a tokenizer and sequence classification model. Assumes the
+        tokenizer and model are HuggingFace transformer-based and on
+        the same model_path
+
+        Args:
+            model_path: str
+                Path to the model to be loaded.
+
+        Returns:
+            SequenceClassification
+                Instance of this class built from the on disk model.
+        """
+        tokenizer = AutoTokenizer.from_pretrained(model_path)
+        model = AutoModelForSequenceClassification.from_pretrained(model_path)
+        device = SequenceClassification._get_device()
+        return cls(
+            tokenizer=tokenizer,
+            model=model,
+            device=device,
+        )
+
+    ################################## Private Functions #########################################
+
+    def _get_scores(self, text: Union[str, List[str]]):
+        """Run tokenizer and model to get scores on text(s)
+
+        Args:
+            text: Union[str, List[str]]
+                Input string(s) to be used
+
+        Returns:
+            scores_dict
+                Dict with key label, and values as the array of scores,
+                each corresponding to text(s)
+        """
+        # Apply tokenizer
+        tokenized_text = self.tokenizer(
+            text,
+            padding="max_length",
+            max_length=self.tokenizer.model_max_length,
```

Comment:
true I suppose this might be unnecessary since `padding='max_length', truncation=True` should truncate and pad to max model input length ref: https://huggingface.co/docs/transformers/pad_truncation